source/common/upstream/upstream_impl.cc

```diff
@@ -99,19 +118,22 @@ Network::ClientConnectionPtr
 HostImpl::createConnection(Event::Dispatcher& dispatcher, const ClusterInfo& cluster,
                            Network::Address::InstanceConstSharedPtr address,
                            const Network::ConnectionSocket::OptionsSharedPtr& options) {
-  Network::ConnectionSocket::OptionsSharedPtr cluster_options;
-  if (cluster.features() & ClusterInfo::Features::FREEBIND) {
-    cluster_options = std::make_shared<Network::ConnectionSocket::Options>();
+  Network::ConnectionSocket::OptionsSharedPtr connection_options;
+  if (cluster.clusterSocketOptions() && !cluster.clusterSocketOptions().get()->empty()) {
```

Comment:
if (cluster.clusterSocketOptions() != nullptr && !cluster.clusterSocketOptions()->empty()) {

Reply:
Can be simplified to just the nullptr check if parseClusterSocketOptions() returns nullptr if the options are empty, as you suggest.